lms/resources/_js_config/__init__.py

```diff
@@ -538,3 +539,32 @@ def _sync_api(self):
             return self._blackboard_sync_api()
 
         return None
+
+    def _add_deeplinking_api(self):
+        """
+        Add the details of the "Deeplinking API" in LMS where we support deeplinking.
+
+        This API will be used by the frontend to retrieve the form fields required
+        for the deeplinking submission while on CONTENT_ITEM_SELECTION mode to store the
+        selected content on the LMS.
+        """
+        if (
+            not self._context.is_canvas
+            or self._application_instance.lti_version != "1.3.0"
```

Comment:
Only 1.3 to split the PRs, this one adding the new API and FE behavior.

The next one using it on LTI1.1 (which will remove some FE code)